### DISCUSSION TITLE
Application profiles

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#165] [Q&A] [answered]
More love for "_preset.pref_" files.
Button **PRESETS**: Import & Export.
From & To.
Android/media/gr.talent.cruiser/Presets.
Can contain multiple "_preset.pref_" files.

**Comment by devemux86** [comment]
@0709wiwiwi thanks for the idea!

"Profiles" would be more suitable and recognized (?) by the users.

We will need a manager, as users should not required to use files.
App can handle the profile management internally with an interface.

**Reply by airald** [reply]
I manage presets of all applications that allow it to keep my personal config in a safe place. I usually save preset files with a timestamp to have versions.

**Comment by MotoUKRider** [comment]
I like the idea of this - certainly it might be useful if you wish to use Cruiser with a totally different GUI/config setup on a regular basis with faster switching than the "Preferences" import/export (which is more Power User, in my opinion).

Putting my "new user" hat on - I feel like an in-app manager would be better for the load/save/import/export unless any of the popular Cloud/Dropbox options would require the user to navigate "around the folders" using the in-built Android file system.

From a UX point-of-view, if any kind of activity can be handled "in app" with the Cruiser style/theme then it makes for a better product.

**Comment by 0709wiwiwi** [comment]


**Reply by devemux86** [reply]
@0709wiwiwi I hope we can find nice translations in the other languages! :slightly_smiling_face:

**Reply by devemux86** [reply]
Another possibility it to use another name in [app profiles](https://github.com/devemux86/cruiser/discussions/261) and use the "profiles" name here.

**Comment by devemux86** [comment]
Would it be possible to save the BRouter profile in the preferences?

_Originally posted by @Leeb64 in https://github.com/devemux86/cruiser/discussions/266#discussion-5467073._

---

Preference files save everything, app settings, BRouter profiles / no-go areas, etc.

Currently when loading preference files, BRouter profiles / no-go areas were not restored.
(only the app settings, because of some Android restrictions)

This will work properly in next Cruiser: **everything** will be restored from the preference files.

**Comment by devemux86** [comment]
"**Profile manager**" is one of the next big features that are currently being prepared in Cruiser.

- Open profile manager in the main menu
- Tap to switch profile, drag to reorder, swipe to delete
- Long press for the profile menu:
  - Create new profiles based on existing ones
  - Import / export your profiles
  - Reset the profiles to their default settings
  - Base profiles (with star) cannot be deleted
- Confirm a profile switch with OK (may restart the app)
- We have a new profile format, that is also easier for reports
- Profile selection / import / export in settings will be removed

You can manage your app profiles on one screen and switch between profiles easily with just 1 click.
Profiles now preserve all app settings and user preferences.

<img src="https://github.com/devemux86/cruiser/assets/3484020/a75e6f76-7595-4957-835b-2415da0d5362" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/a16cffe6-564b-4809-895c-580c3e691f43" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/17bf4bd9-276f-402f-9699-b37f75b89d8a" width="33%"/>

**Comment by airald** [comment]
I suppose that .pref file should disappear and use the .json files to save preferences. So in this situation, a complete backup will need one json for each profile. It could be more difficult (or take more time for users) but allow the ability to share profiles. 

Maybe the new concept of the "old" import/export could be to save all (or selected) profiles in a folder, and something similar to import. 

If this is complicate or too much work, save the used profiles is not complicate and just take less than a minute.

json files are human readable and allow a simple user to compare 2 profiles easily, I think is a good idea the new format. There are more than 300 settings saved on each profile, this is a high level of customisation!

**Reply by devemux86** [reply]
> I suppose that .pref file should disappear and use the .json files to save preferences.

Correctly, although we can keep import of .pref files for a period of time to allow users to migrate easily to the new profile manager.

> a complete backup will need one json for each profile.

I can implement an "Export all" that exports all profiles to a zip file (like with favorites).

> json files are human readable and allow a simple user to compare 2 profiles easily, I think is a good idea the new format.

@airald :+1: 

> There are more than 300 settings saved on each profile, this is a high level of customisation!

Cruiser customization is indeed insane! :slightly_smiling_face:

**Comment by Ciutnik** [comment]
Looking through the .json file, I see a record of the type `"avoidMotorwaysFactor" : 1,`. Does this mean that we have an influence on the power of avoidance? 

@airald 
> There are more than 300 settings saved on each profile, this is a high level of customisation!

When you look closely, many of them seem to be related to other profiles. For example, a motorcycle profile and the parameters of hiking trails are described... Which of course doesn't change the fact that 

> Cruiser customization is indeed insane! 🙂

And for that, many of us love him ❤️

@devemux86 
> I can implement an "Export all" that exports all profiles to a zip file (like with favorites).

Maybe the "Import all" function to complete? Also for Favorites it would make sense if icons, colors and visibility on the map/route were also imported 🙂

**Reply by devemux86** [reply]
> Looking through the .json file, I see a record of the type "avoidMotorwaysFactor" : 1

This is a remnant from the past when we were testing levels in BRouter avoidances.
We may re-introduce them, we just need to better understand how their levels work.

> Maybe the "Import all" function to complete?

Import in profiles (and favorites) allows multiple files to be selected.
We can import multiple files of profiles (or favorites) simultaneously.

**Comment by devemux86** [comment]
(now that all reports have been resolved)

Please test the new profile manager very well and let me know if we need anything else:
- https://github.com/devemux86/cruiser/discussions/299

**Comment by airald** [comment]
I see the new option to export all profiles to a ZIP file. It works, and the name is the timestamp, useful to have versions.

If I'm not wrong, to import, we need to select which profile import with long tap and then select the uncompressed file stored into the zip. For me it's OK, better than import all profiles at the same time. There's no need to have the same names from new "container" profile and backup profile.

**Reply by devemux86** [reply]
In import, if we select multiple profile files, they are imported as new profiles.
If we select one profile file during import, then it updates the selected profile.

Note: Base profiles (with star) can be updated except for their name and type.

**Reply by airald** [reply]
I don't found the option to import except updating an existing profile.

**Reply by devemux86** [reply]
@airald the context menu appears when long press on any profile.
Create a new profile and import a profile file to update its settings.

**Reply by devemux86** [reply]
@airald 

Or when we import a profile we can show a question if it should create a new profile or update the existing one.

**Reply by airald** [reply]
updating an existing profile is the only way I found to import, and I think is enough, we usually have few profiles. Using this way it's possible import a friend profile into a new profile. i.e: I can import a friend profile with a name of one of my existing prfiles without overwrite it.

I usually makes backups, so it's very useful to backup all profiles at the same time, but only import when something is wrong or in special situations.

**Comment by Ciutnik** [comment]
I'm just thinking... 
Since we have unimaginably configurable profiles, we can create and modify them freely, we already have everything to define different types of traveling within one route, by combining different profiles.

Having an "Urban Motorcycle" profile (prefers fast roads but not highways), a "Country Motorcycle" profile (only tertiary and gravel roads), and a "Road Motorcycle" profile (prefers highways), we will easily create a dream trip. That is three segments: shortest and fastest to the borders of the city, then a pleasant trip through the backcountry, away from the cities, and finally back on the highway for dinner. 

What's more! With the "Hiking" profile, we can plan a walk in the forest in the meantime (I know, it's uncomfortable with a helmet and moto boots 😀). This is a super universal solution that is a dream come true! By adding even a walk, it is even more than we know and remember 🙂

Well, it's probably not that simple. For example, the fact that changing the profile requires restarting the application... But it's nice to dream 🙂

**Reply by devemux86** [reply]
> the fact that changing the profile requires restarting the application

This is only needed for a few core settings, e.g. app language, theme, etc.
When we change settings with the normal way, only some restart the app.

Switching profiles currently needs a restart, because there is no such separation.
But I could check these settings in the profiles and restart only when it is needed.

e.g. switch profiles that have different BRouter options, it should not require restart.

**Reply by Ciutnik** [reply]
So it would be possible to change the parameters (profile) of the BRouter and even the router within one route? This would be another game changer in Cruiser development 🙂 This would probably be the last element of K2 that many miss.

**Reply by devemux86** [reply]
@Ciutnik we are probably talking about different things?

This is about the app profiles: travel type + app settings.

What you mean is the routing profiles? It is a different feature:
- https://github.com/devemux86/cruiser/discussions/142

**Reply by Ciutnik** [reply]
> What you mean is the routing profiles? It is a different feature

Yes, but from the user's point of view, the end result is the same. Routing profiles (i.e. a set of routing preferences, including the router) are already included in the application profiles, so maybe routing profiles wouldn't be necessary? That's why I said about super-universal solution.

**Reply by devemux86** [reply]
@Ciutnik correctly, all routing (and more) options are included in the app profiles.

You mentioned different routers or options in one route.
Currently this is available in the navigation re-routing.

**Reply by Ciutnik** [reply]
> Currently this is available in the navigation re-routing.

But not in planning. In planning, we can choose either what we have set in the active profile, or a straight line. What if it was possible to indicate a ready-made application profile as a source of routing parameters? All the necessary data is included in it.

**Reply by devemux86** [reply]
@Ciutnik 

> What if it was possible to indicate a ready-made application profile as a source of routing parameters?

Something like that is already planned in:
- https://github.com/devemux86/cruiser/discussions/142

We'll see if it can be combined with the app profiles, that contain much more data,
or it can be offered with another way. This a large and very interesting feature. :slightly_smiling_face:

**Comment by devemux86** [comment]
A great new feature is coming to app profiles:
- We can now **switch profiles instantly** _without restarting the app_

I changed the profile loading mechanism (there are very few settings that need restart).
e.g. app theme, language, color

**Reply by devemux86** [reply]
> We can now **switch profiles instantly** _without restarting the app_

Available for testing in:
- https://github.com/devemux86/cruiser/discussions/299

**Reply by airald** [reply]
Nice, much clean!
Just can be better if you switch profile while navigating, then starts the navigation again.

**Comment by devemux86** [comment]
Another interesting feature is that each profile could only display its own settings.

e.g. "Motorcycle" profiles show only the motorcycle options for BRouter, speeds, voice distances, etc.
Similarly, "Car" or "Bike" or "Hiking" profiles show only the settings of their types.

This would greatly simplify the settings, so users don't see options of other profiles.

**Reply by airald** [reply]
I see in "main menu" only appears the selcted profile type options but in settings appear all in order to change settings of any profile type without need to change into many profiles. Very useful to keep main menu more clean.

**Reply by devemux86** [reply]
> Another interesting feature is that each profile could only display its own settings.

All profile settings have been reorganized and simplified in:
- https://github.com/devemux86/cruiser/discussions/299

---

It would be more useful if the BRouter (offline) options were more _directly accessible_.
Maybe move them directly into the routing options or it wouldn't be nice if they are separate?

Remember: profile routing options are directly accessible in the menu "Routes | Options".

---

App profiles should now be considered complete and ready for public release.
Please test them very well and check if everything works properly!

**Comment by devemux86** [comment]
More profile improvements are available in:
- https://github.com/devemux86/cruiser/discussions/299

App language is now global and there are more checks for the (global) storage folder.

---

Please _**try to break the profiles**_!

Import other profiles as `pref` and `json` files, with different storage folder, app language, active maps / poi,
try to make any possible change and test you can imagine. :slightly_smiling_face:

If you see anything strange, please let me know and try to remember your steps.

**Reply by devemux86** [reply]
Also make a clean install of Cruiser from Google Play on a device that is not installed,
make changes and then install on top the Cruiser (Beta) to see if everything works ok.

**Comment by airald** [comment]
Active map is global?

**Reply by devemux86** [reply]
Global: active map / poi, app language, storage folder.

**Comment by devemux86** [comment]
With the new [profile manager](https://github.com/devemux86/cruiser/discussions/165#discussioncomment-6800089), I am thinking to add more profile types:
- **Running**
- ...

Do you think we need more or something related to bikes or motor vehicles?

**Reply by WolfgangZ42** [reply]
A second motorcycle profile would I appreciate: "Motorcycle offroad"! "Motorcycle" for onroad tours and "Motorcycle offroad" for gravel tours. You then are able to set different zooming level and annoucement distances, etc. 
Actually I'm using the "Mountain bike" profile for that - works!

**Reply by devemux86** [reply]
@WolfgangZ42 

We can already create extra profiles based on the default (or custom) profiles and customize them differently.
(long press on profiles and select "New profile")

I am just thinking here about adding some new profile types (with icons) that may be useful in different needs:
running, racing bike, moped, etc.

"Running" is certainly needed. I am still not sure about the others, as they can be covered by existing profiles.

**Reply by WolfgangZ42** [reply]
> We can already create extra profiles 

Didn't remember that! Now it took me few minutes to have this: 
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/dc4d3372-6bfe-4065-b8a9-628b6298f2b1"/>
Perfectly implemented - my compliments!

**Reply by devemux86** [reply]
@WolfgangZ42 :+1: 

- **Default profiles** (with the star): for obvious reasons cannot delete or change their default name / type.

- **Custom profiles**: we can create new extra profiles with any name / type based on all existing profiles.

We can customize the settings of all profiles as much as we want.

**Comment by devemux86** [comment]
Alternative implementations without (hidden) long press menu. Which do you prefer?
- Drag up / down to reorder (with right indicator)
- Swipe left / right to delete
- **Tap right indicator for menu**
- Tap anywhere to activate profile

<img src="https://github.com/devemux86/cruiser/assets/3484020/4ac15f6d-b406-411c-8471-b7d5d6f29e5d" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/16d50060-2ae6-47b0-bf07-86c8c9d38de1" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/bdc3d122-c644-47e4-aaba-60793c59721f" width="33%"/>

**Reply by Ciutnik** [reply]
Wow! Both look great! It seems to me that the first one (two parallel lines) can be understood as just only for change of order. I think second one (six dots) can be better understood as the different activities associated with it. This is my first impression 🙂

**Reply by devemux86** [reply]
Similar impression here, that's why both images. Android also uses the "=" icon for reorder in system lists.
We could use the triple line, like in the app menu button, but probably it could be too much?

I will see how this can be implemented in the other lists with similar workflow: favorites, waypoints, search.

**Reply by devemux86** [reply]
I updated the [above images](https://github.com/devemux86/cruiser/discussions/165#discussioncomment-6949064) with the triple line example.

**Reply by Ciutnik** [reply]
Initially, I imagined that the "hamburger menu" would not be associated much with changing the order... but I really like it a lot. Thanks for the illustrations!

While "double line" are associated with a vertical change of order, my initial favorite "three dots" may be associated with horizontal dragging. However, I vote for "triple line" 😀

**Reply by devemux86** [reply]
It is one of the Material drag indicators: https://fonts.google.com/icons?icon.set=Material+Icons&icon.query=drag

**Reply by devemux86** [reply]
If search for menu, both 2-line and 3-line icons are in results: https://fonts.google.com/icons?icon.set=Material+Icons&icon.query=menu

**Reply by devemux86** [reply]
You can test the implementation in profiles in:
- https://github.com/devemux86/cruiser/discussions/299

**Reply by DolomitesTom** [reply]
When tapping right I don't get any menu to maintain the profile. The six dots do nothing neither on tapping nor on long pressing.

**Reply by Ciutnik** [reply]
@DolomitesTom 
I'm guessing you have a Samsung? 🙂

**Reply by devemux86** [reply]
@DolomitesTom Samsung probably modified the default Android timers. :slightly_smiling_face: 

It was also reported [here](https://github.com/devemux86/cruiser/discussions/299#discussioncomment-6949977), there will be a new Cruiser later today to fix this.

**Reply by DolomitesTom** [reply]
Yes, 2x Samsung, OK thx

**Comment by devemux86** [comment]
A different implementation to fix the problems in Samsung devices:
- https://github.com/devemux86/cruiser/discussions/299

@airald @Ciutnik @DolomitesTom and others, please test in your Samsung devices.

---

- Long press anywhere and drag up / down to reorder
- Swipe left / right to delete
- **Tap right indicator for menu**
- Tap anywhere to activate profile

**Reply by DolomitesTom** [reply]
Works now, thx

**Reply by devemux86** [reply]
The alternative is to show the menu _also_ with long press
and use only the left icon for the drag to reorder gesture.

Like the favorites / waypoints lists work now (which will be similar to profiles list).

I am not sure which one is better and more user friendly?

**Reply by Ciutnik** [reply]
I confirm that the "six dots" menu works properly on my two Samsung tablets. Thanks! 👍

By the way, since you plan to release the public Beta soon, it's probably time to decide on the naming. This is mentioned at the beginning of the topic. Shouldn't "Profiles" be replaced with "Presets" to reserve "Profile" exclusively for routing options?

**Reply by WolfgangZ42** [reply]
> Like the favorites / waypoints lists work now (which will be similar to profiles list).

I would prefer the same behavior in waypoint-, favorites- and profile-lists.

**Reply by devemux86** [reply]
> it's probably time to decide on the naming.

Better to keep the term "Profile" as it's a more common word, easily translated to Cruiser's 10 languages(!)
and users also know it from other map apps.

Note: there is no more a profile option in the route settings.
This is set automatically from the app profile which contains all app settings.

We can change the "Type" of custom profiles with the profile menu.

> I would prefer the same behavior in waypoint-, favorites- and profile-lists.

I am currently adapting all these lists to work similarly. :slightly_smiling_face: 
This was just a quick Beta to fix the Samsung issue.

**Comment by devemux86** [comment]
<img src="https://github.com/devemux86/cruiser/assets/3484020/295c7d91-8229-4322-ad1f-79573500a409" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/3a4a7718-1556-4312-98d9-e8e6ac2ccde7" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/c75713b5-4274-4547-8adc-02d6bb5e3467" width="33%"/>

**Reply by Ciutnik** [reply]
I hadn't thought about that before. The list of points on the route (picture 1) is probably a good argument for making it easier to change the order. That is, the option with the left icon as a handle, not a long tap.

**Comment by devemux86** [comment, marked as answer]
New profile manager is available in [Cruiser 3.0.32](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by devemux86** [comment]
I am not a fan of such "hidden features" (like the current profile manager), I prefer when everything is immediately and directly recognizable and accessible without the need to "tap somewhere to check if there is some functionality hidden behind"...

_Originally posted by @bios4 in https://github.com/devemux86/cruiser/discussions/35#discussioncomment-7078803._

**Reply by devemux86** [reply]
It would be nice to put the Profiles as a large button.

We also have to think about symmetry (6 or 8 buttons):
- Add buttons for Profile + Subscriptions
- Replace Tools button with Profile button
  - Then we have to move the tools content elsewhere

**Reply by bios4** [reply]
From my point of view you can easily add the profile manager plus the subscriptions as two additional big buttons!
This should fit a recent smarphone screen without problems...

**Reply by Ciutnik** [reply]
In my humble opinion, the main menu header, where the name of the current profile is displayed, is a good "button" to activate the profile manager. That is, as it is now. 

I would only make it less hidden by adding an icon on the right side (six dots that we already use elsewhere) and make this header higher - just like the footer with the buttons "Subscription" and "X".

**Reply by devemux86** [reply]
> make this header higher - just like the footer with the buttons "Subscription" and "X".

This is easy! :+1:

**Comment by 0709wiwiwi** [comment]
These app profiles function are usefully fine. However each minor change during an operation is immediately saved in the current profile.
Would it not be possible to add a lock (open/close) this to protect the profiles a little bit for such unexpected or accidental changes ?

**Reply by devemux86** [reply]
App settings (stored in app profiles) can only be changed by users in an intentional (not unexpected) action.

A lock switch will be again handled by the users. Or do you mean specifically the routing options?

**Reply by devemux86** [reply]
Android saves settings automatically, anything else requires more and complicated work.
So need to think carefully if and how it is possible technically to offer something different.

**Comment by GuG1050** [comment]
I am preparing a trip to Greece and  experimented a bit with profile ex/import to synchronize my 3 devices Pixel 6 (on the bike, navigation only), Pixel 7pro (my phone), RealmePad (for route planning in the hotel). 
I set up Cruiser on Pixel 6 as master. Exported the Motorbike profile and imported in the two other devices. Now my surprise: Where the two Pixels now calculate exactly the same route, the RealmePad calculates someting completely different for the first third of the Route, the rest is the same. I did the ex-import several times, didnt touch anything on the Realme, no change. Any idea why this could happen? Maps and Brouter data are up to date and the same on all devices, version is 3.0.42.

Pixels | Realme
<img src="https://github.com/devemux86/cruiser/assets/122791327/875d179c-0b78-45f7-a155-c2db89e3363c" width="40%"/><img src="https://github.com/devemux86/cruiser/assets/122791327/d884fcbe-34d2-4b6d-91f3-4c96456af615" width="50%"/>

**Reply by devemux86** [reply]
@GuG1050

Have you exported a profile as json, transfer it to the other device and import it?
What have you selected during import? "New profile" or "Update profile"?

To "Update profile" have you pressed the right menu of the correct profile to update?

What router do you use, BRouter (offline)? Internal or external?
If using external BRouter app, is it updated to the latest version?

Can you post the profile file here to check it?

<img src="https://github.com/devemux86/cruiser/assets/3484020/717dcb1c-3a87-4190-a211-45acd889e910" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/2605d105-d7f1-4e06-ba6e-611c1ff5de67" width="50%"/>

**Reply by GuG1050** [reply]
Have you exported a profile as json, transfer it to the other device and import it? -> Yes 
What have you selected during import? "New profile" or "Update profile"? -> Update profile

To "Update profile" have you pressed the right menu of the correct profile to update?  -> Yes

What router do you use, BRouter (offline)? Internal or external? -> internal
If using external BRouter app, is it updated to the latest version? -> n/a

Can you post the profile file here to check it? -> posted
[GGMotorrad.json](https://github.com/devemux86/cruiser/files/14862823/GGMotorrad.json)

**Reply by devemux86** [reply]
@GuG1050 

Are you sure that the same app profile is selected (is colored) in both devices?
Can you check the BRouter routing options in each device, are they the same?

How do you calculate the route? Import a GPX track or other format?

**Reply by devemux86** [reply]
@GuG1050 

Or was the route calculated externally with another router,
so if recalculate it in the 1st device it will change there too?

**Reply by devemux86** [reply]
@GuG1050 can you post or send the route file?

**Reply by bios4** [reply]
> Can you check the BRouter routing options in each device, are they the same?

Aren't they part of the exported profile and therefore surely match after importing on the 2nd device?

**Reply by devemux86** [reply]
@bios4 correctly, but we need to be sure that:

- Both devices have active the same app profile
e.g. not update a profile and have active another one

- Route is calculated inside the app in same way
e.g. not import an external route calculated elsewhere

**Reply by GuG1050** [reply]
Are you sure that the same app profile is selected (is colored) in both devices?  > Yes
Can you check the BRouter routing options in each device, are they the same?  > Yes

How do you calculate the route? Import a GPX track or other format?  > No, only within Cruiser

> @GuG1050 can you post or send the route file?   > How can I send the file? What Format?

**Reply by devemux86** [reply]
@GuG1050 export the route as json file and attach it here.

**Reply by devemux86** [reply]
@GuG1050 also make sure that BRouter routing data are similarly updated in all devices.
(menu Routing | BRouter)

In case data changes exist in OpenStreetMap that only the updated device received them.

**Reply by GuG1050** [reply]
also make sure that BRouter routing data are similarly updated in all devices.
(menu Routing | BRouter).   > Done on all 3 devices in parallel yesterday. Did it again a few minutes ago. No change.

RealmePad Route:

[20240404_115425.json](https://github.com/devemux86/cruiser/files/14864743/20240404_115425.json)

**Reply by 0709wiwiwi** [reply]
Advise: Send as well the result files from your first and the third device in  .json or .kurviger or gpx navtrk format to Emux

**Reply by GuG1050** [reply]
Here the files from the two Pixels.

[P6-20240404_121909.json](https://github.com/devemux86/cruiser/files/14864934/P6-20240404_121909.json)
[P7pro-20240404_121810.json](https://github.com/devemux86/cruiser/files/14864935/P7pro-20240404_121810.json)

**Reply by devemux86** [reply]
@GuG1050 

Calculating the route here with your profile, it also has as a result the Pixel route.
(blue: imported Realme route, magenta: calculated route)

Is it possible to post screenshots from the 2 devices with all BRouter settings (all screens)?

<img src="https://github.com/devemux86/cruiser/assets/3484020/9b489649-45fa-461e-b5d3-657bf4535138" width="50%"/>

**Reply by GuG1050** [reply]
Here the BRouter screenshots 3 screens from 3 devices. Sorry, I didnt know how to resize 🥴.

<img src="https://github.com/devemux86/cruiser/assets/122791327/b682e0fd-596a-4925-88d3-0f6797797145" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/122791327/86051015-da6f-43bc-af26-67cc5f5a4383" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/122791327/c70f4e0b-90dd-4412-ae8e-b8aa574e306c" width="33%"/>
<img src="https://github.com/devemux86/cruiser/assets/122791327/0e8626ce-9c1a-4e39-b53e-028aef8efeab" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/122791327/2bd7f5fa-0a86-4f87-aa90-55c9f853d014" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/122791327/51b87190-2949-42f6-b359-023ab8ce0f46" width="33%"/>
<img src="https://github.com/devemux86/cruiser/assets/122791327/88de1e0b-56e1-4898-97ab-3aee07ba288e" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/122791327/224ba716-ec36-4353-88f6-89bc323a0146" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/122791327/ab140236-9a2d-467e-81a3-6433421fff86" width="33%"/>

**Reply by GuG1050** [reply]
RealmePad continues to calculate the blue route ...

**Reply by devemux86** [reply]
@GuG1050 thanks very much for the screenshots!

These are exactly what I also see here when I import your profile.

Can you **move** the end point and recalculate the route in the Realme and check the result?

**Reply by GuG1050** [reply]
Moved end point to the intersecction with A12. Only a few 100 meters . Now route ist the same also on RealmePad. Interesting.  
I would have expected a difference not in the very beginning but in the end of a 450km route when I move the end point only a little.
Does this mean when e.g. planning a route with a friend on 2 devices it is mandatory to place the end point EXACTLY the same on both devices as minor differences can cause a completly different route at the start?

**Reply by devemux86** [reply]
Generally the waypoints should be on the same locations to make exactly the same tests in both devices.
e.g. export route in one device, transfer route to the other device, recalculate route: should be same route.

But here the difference is at the start. I asked to move slightly the end point because:

---

But in this case I think it is the cache of BRouter. If start + end and options are the same,
new calculation returns the cached route. Somehow in your case there is a cached route.

Moving start or end forces a clean route calculation.
Cache can be disabled, but have not seen a problem and it is always useful with section profiles.

**Reply by devemux86** [reply]
Share routes (and other Cruiser data) with friends will be much simpler in the next version.

There will a new "Share" action in all menus and it will work offline without internet or cloud:

- https://github.com/devemux86/cruiser/discussions/330#discussioncomment-8782827

**Reply by GuG1050** [reply]
Thank you for your effort. I have no problem with these implications. Good to know that profile ex-import ist not an issue and how to take corrective action for route calculations. 
I am looking forward to the new sharing functionalities.

**Reply by 0709wiwiwi** [reply]
> Moving start or end forces a clean route calculation.

A recalculate all (this example has only start and end) could have had the same effect or not  ?

**Reply by devemux86** [reply]
> A recalculate all (or in this case has only start and end even a recalculate section) could so have had the same effect ?

Currently the cache is used in all actions, I will check it again.
